content/sensu-go/5.17/guides/secrets-management.md

```diff
@@ -0,0 +1,289 @@
+Replace `INTEGRATION_KEY` with your PagerDuty Events API Integration Key.
+
+This writes your secret into Vault.
+In this example, the name of the secret is `pagerduty`.
+The `pagerduty` secret contains a key, and you specified that the `key` value is your PagerDuty Events API Integration Key.
+The `id` value for your secret will be `secret/pagerduty#key`.
+
+The `id` value for secrets that target a HashiCorp Vault must start with the name of the secret's path in Vault.
+The Vault dev server is preconfigured with the `secret` keyspace already set up, so we recommend using the `secret/` path for the `id` value while you are learning and getting started with Vault secrets management.
+
+Run `vault kv get secret/pagerduty` to see the secret you just set up.
+
+Use `sensuctl create` to create your `vault` secret:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: pagerduty_key
+  namespace: default
+spec:
+  id: secret/pagerduty#key
+  provider: vault
+EOF
+{{< /highlight >}}
+
+Now you can securely pass your PagerDuty Events API Integration Key in Sensu checks, handlers, and mutators by referring to the `pagerduty_key` secret.
+In this guide, you'll use your `pagerduty_key` secret in a handler.
+
+## Add a handler
+
+### Register the PagerDuty Handler asset
+
+To begin, register the [Sensu PagerDuty Handler asset][23] with [`sensuctl asset add`][22]:
+
+{{< highlight shell >}}
+sensuctl asset add sensu/sensu-pagerduty-handler:1.2.0 -r pagerduty-handler
+{{< /highlight >}}
+
+This example uses the `-r` (rename) flag to specify a shorter name for the asset: `pagerduty-handler`.
+
+_**NOTE**: You can [adjust the asset definition][26] according to your Sensu configuration if needed._
+
+Run `sensuctl asset list --format yaml` to confirm that the asset is ready to use.
+
+With this handler, Sensu can trigger and resolve PagerDuty incidents.
+However, you still need to add your secret to the handler spec so that it requires your backend to request secrets from your secrets provider.
+
+### Add your secret to the handler spec
+
+To create a handler definition using Sensu's built-in `Env` secret provider, run:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+api_version: core/v2
+type: Handler
+metadata:
+  namespace: default
+  name: pagerduty
+spec:
+  type: pipe
+  command: pagerduty-handler
+  secrets:
+  - name: pagerduty_key
+    secret: env
+  runtime_assets:
+  - pagerduty-handler
+  timeout: 10
+  filters:
+  - is_incident
+EOF
+{{< /highlight >}}
+
+To create a handler definition using your HashiCorp Vault secret, run:
+
+{{< highlight shell >}}
+cat << EOF | sensuctl create
+---
+api_version: core/v2
+type: Handler
+metadata:
+  namespace: default
+  name: pagerduty
+spec:
+  type: pipe
+  command: pagerduty-handler
+  secrets:
+  - name: secret/pagerduty#key
+    secret: vault
+  timeout: 10
+  filters:
+  - is_incident
+EOF
+{{< /highlight >}}
```

Comment:
The handler definition will be the same regardless of the secret provider. This second can be removed.